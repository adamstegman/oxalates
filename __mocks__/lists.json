[
  {
    "id": "all",
    "name": "All",
    "bottomThreshold": null,
    "topThreshold": null
  },
  {
    "id": 1,
    "name": "Very High",
    "bottomThreshold": 50,
    "topThreshold": null
  },
  {
    "id": 2,
    "name": "High",
    "bottomThreshold": 15,
    "topThreshold": 50
  },
  {
    "id": 3,
    "name": "Moderate",
    "bottomThreshold": 5,
    "topThreshold": 15
  },
  {
    "id": 4,
    "name": "Low",
    "bottomThreshold": null,
    "topThreshold": 5
  }
]
