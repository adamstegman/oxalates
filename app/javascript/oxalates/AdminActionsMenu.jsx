import React from 'react';
import PropTypes from 'prop-types';

import { FoodActionsMenu } from './FoodActionsMenu';

export class AdminActionsMenu extends React.Component {
  render() {
    const {
      activeListId,
      authenticated,
      editingFoods,
      setEditingFoods,
      startNewFood,
    } = this.props;

    if (authenticated) {
      return <FoodActionsMenu activeListId={activeListId}
                              editingFoods={editingFoods}
                              setEditingFoods={setEditingFoods}
                              startNewFood={startNewFood} />;
    }

    return null;
  }
};

AdminActionsMenu.propTypes = {
  activeListId: PropTypes.node,
  authenticated: PropTypes.bool,
  editingFoods: PropTypes.bool,
  setEditingFoods: PropTypes.func.isRequired,
  startNewFood: PropTypes.func.isRequired,
};
