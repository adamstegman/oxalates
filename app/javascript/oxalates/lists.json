[
  {
    "id": "all",
    "name": "All",
    "bottom_threshold": null,
    "top_threshold": null
  },
  {
    "id": 1,
    "name": "Very High",
    "bottom_threshold": 50,
    "top_threshold": null
  },
  {
    "id": 2,
    "name": "High",
    "bottom_threshold": 15,
    "top_threshold": 50
  },
  {
    "id": 3,
    "name": "Moderate",
    "bottom_threshold": 5,
    "top_threshold": 15
  },
  {
    "id": 4,
    "name": "Low",
    "bottom_threshold": null,
    "top_threshold": 5
  }
]
