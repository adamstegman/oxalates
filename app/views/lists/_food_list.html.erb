<ul class="food-list">
<% foods.each do |food| %>
  <li class="food-list-item <%= list_class_name(food.list) %>"><%= food.name %></li>
<% end %>
</ul>
