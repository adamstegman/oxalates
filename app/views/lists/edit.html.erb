<%= form_tag list_path(@list.id), method: :put do %>
  <ul class="food-list">
  <% @list.foods.each do |food| %>
    <li>
      <%= link_to food.name, edit_food_path(food), class: "edit-food-list-item #{list_class_name(food.list)}" %>
      <%= hidden_field_tag 'list[foods][][id]', food.id %>
      <%= label_tag 'list[foods][][name]', "Name", class: 'food-list-item-name' %>
      <%= text_field_tag 'list[foods][][name]', food.name, class: 'food-list-item-name' %>
    </li>
  <% end %>
  </ul>
<% end %>
