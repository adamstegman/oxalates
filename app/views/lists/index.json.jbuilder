json.lists(@lists) do |list|
  json.(list, :id, :name, :bottom_threshold, :top_threshold)
end
