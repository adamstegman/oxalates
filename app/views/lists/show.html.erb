<%= render "food_list", foods: @list.foods %>
