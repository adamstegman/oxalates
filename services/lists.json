[
  {
    "name": "Very High",
    "bottomThreshold": 50
  },
  {
    "name": "High",
    "bottomThreshold": 15,
    "topThreshold": 50
  },
  {
    "name": "Moderate",
    "bottomThreshold": 5,
    "topThreshold": 15
  },
  {
    "name": "Low",
    "topThreshold": 5
  }
]
