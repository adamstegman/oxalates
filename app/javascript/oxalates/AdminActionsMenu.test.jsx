import React from 'react';
import renderer from 'react-test-renderer';

import lists from './__mocks__/lists.json';
import { AdminActionsMenu } from './AdminActionsMenu';

test('AdminActionsMenu renders admin action buttons', () => {
  const component = renderer.create(
    <AdminActionsMenu authenticated={true}
                      activeListId={lists[0].id}
                      editingFoods={false}
                      setEditingFoods={() => {}}
                      startNewFood={() => {}} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('AdminActionsMenu renders nothing', () => {
  const component = renderer.create(
    <AdminActionsMenu authenticated={false}
                      activeListId={lists[0].id}
                      editingFoods={false}
                      setEditingFoods={() => {}}
                      startNewFood={() => {}} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
