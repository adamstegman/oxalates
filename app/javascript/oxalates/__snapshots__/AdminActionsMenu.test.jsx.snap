// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminActionsMenu renders admin action buttons 1`] = `
<div
  className="actions-container"
>
  <button
    className="food-action"
    onClick={[Function]}
    type="button"
  >
    Add
  </button>
  <button
    className="food-action"
    onClick={[Function]}
    type="button"
  >
    Edit
  </button>
</div>
`;

exports[`AdminActionsMenu renders nothing 1`] = `null`;
